core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -620,6 +620,14 @@ private[spark] class TaskSetManager(
     // Note: "result.value()" only deserializes the value when it's called at the first time, so
     // here "result.value()" just returns the value and won't block other threads.
     sched.dagScheduler.taskEnded(tasks(index), Success, result.value(), result.accumUpdates, info)
+    // Kill other task attempts if any as the one attempt succeeded
```

Comment:
Can you change this to "Kill any other attempts for the same task (since those are unnecessary now that one attempt completed successfully)."